library/test/src/test/java/com/bumptech/glide/load/model/StreamEncoderTest.java

```diff
@@ -47,4 +49,38 @@ public void testWritesDataFromInputStreamToOutputStream() throws IOException {
 
     assertEquals(fakeData, new String(data, "UTF-8"));
   }
+
+  @Test(expected = NullPointerException.class)
+  public void testFileNullException() throws IOException {
+    String fakeData = "SomeRandomFakeData";
+    ByteArrayInputStream is = new ByteArrayInputStream(fakeData.getBytes("UTF-8"));
+    boolean success = false;
+
+    // Write some data to #file making sure file.delete() executes properly
```

Comment:
I'm not sure what value this step adds? Could you clarify?

Reply:
I was also confused, below explains what I thought:
* if i do nothing to `file` variable in this method, it would thrown a `IllegalStateException` error from `junit.After.tearDown()` method indicating deleting unsuccessful.
* From my interpretation, when new File() method is called in `junit.Before` method and creates a new `File` object, an actual file will not be created in the system until some data is written to that file. Trying to delete a non-existing file leads to the failure of `delete` mentioned in bullet point 1.
* So I wrote some data to `file` object to make the actual file existing in the system and make sure the `junit.After.file.delete()` method executes properly

Reply:
I think you could simplify by calling [`File.createNewFile()`](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createNewFile()). You can also mention the check in tearDown in a comment explicitly. That way there's no confusion about the encoder being involved and it's clearer that you're doing something just for the test setup.

Reply:
Done